pkg/volmount/getvolumepathnamesforvolumename.go

```diff
@@ -0,0 +1,69 @@
+package volmount
+
+import (
+	"path/filepath"
+	"syscall"
+	"unicode/utf16"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+// utf16ToStringArray returns the UTF-8 encoding of the sequence of UTF-16 sequences s,
+// with a terminating NUL removed. The sequences are terminated by an additional NULL.
+func utf16ToStringArray(s []uint16) (result []string) {
```

Comment:
Interesting, `hcsshim/internal/winapi.ConvertStringSetToSlice` works the other way 'round to my version here, it fails if it doesn't find the expected double-null.

I feel like if they're doing the same thing apart from character-width (that one is for an `...A` call, this one is for a `...W` call), they probably should be consistent, particularly if we want to punt them upstream. Should I flip mine around, or is the 'best effort' semantic useful?

Reply:
Hmm, that's a good question. Do you think having this `best effort` semantic is actually needed anywhere? The syscall description specifically mentions that the buffer will have double-null at the end. And if we don't find a double-null in the buffer it means something has gone wrong. So I prefer throwing an error rather than just swallowing it.

Reply:
I thought I'd seen the `best effort` behaviour somewhere else, but honestly don't recall where, now. Easy-enough to change (I already prototyped copying `ConvertStringSetToSlice` from hcsshim and making it work with `*uint16` in my working copy, so I can just squash that in next time I'm on this.